packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -522,6 +524,21 @@ const initializeProject = async (extensionContext: vscode.ExtensionContext) => {
   registerPushOrDeployOnSave();
   await decorators.showOrg();
 
+  // Initialize metadata XML support
+  const metadataXmlSupport = MetadataXmlSupport.getInstance();
+  await metadataXmlSupport.initializeMetadataSupport(extensionContext);
+
+  // Initialize metadata hover provider
+  const metadataHoverProvider = new MetadataHoverProvider();
+  await metadataHoverProvider.initialize();
+
+  // Register hover provider for XML files
+  const hoverProviderDisposable = vscode.languages.registerHoverProvider(
```

Comment:
Why was this necessary to register a new provider, as opposed to providing the built xsd to the built-in xml extension?

Reply:
via, ex: https://github.com/redhat-developer/vscode-xml/blob/main/docs/Preferences.md#file-associations